core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -176,15 +176,16 @@ public void initialize(String name, Map<String, String> unresolved) {
     this.client = clientBuilder.apply(mergedProps);
     this.paths = ResourcePaths.forCatalogProperties(mergedProps);
 
+    String token = mergedProps.get(OAuth2Properties.TOKEN);
```

Comment:
I think we want to use the `initToken` if no token was sent back. That should be what happens here since it was pulled from the original properties, but it isn't obvious and I think you can remove properties in the config route. What about using `getOrDefault(TOKEN, initToken)` to make it clear?

Reply:
Yeah, it's functionally the same, but if you think that's more clear, I can update it.

Reply:
No, I think it's fine. Not worth waiting on CI.